testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap740/DomainAdjuster740.java

```diff
@@ -48,10 +48,19 @@ protected List<ModelNode> adjustForVersion(final DomainClient client, PathAddres
         adjustRemoting(ops, profileAddress.append(SUBSYSTEM, "remoting"));
         adjustUndertow(ops, profileAddress.append(SUBSYSTEM, "undertow"));
         removeDistributableEjb(ops, profileAddress.append(SUBSYSTEM, "distributable-ejb"));
+        adjustInfinispan(ops, profileAddress.append(SUBSYSTEM, "infinispan"));
 
         return ops;
     }
 
+    private void adjustInfinispan(List<ModelNode> ops, PathAddress subsystem) {
+        // Module org.wildfly.clustering.singleton.server is an alias of org.wildfly.clustering.server
+        // Legacy hosts don't understand org.wildfly.clustering.singleton.server, so change it to what they understand
+        final PathAddress cacheContainer = subsystem.append("cache-container", "server");
+        final ModelNode value = new ModelNode().addEmptyList().add("org.wildfly.clustering.server");
+        ops.add(Util.getWriteAttributeOperation(cacheContainer, "modules", value));
```

Comment:
@yersan @pferraro This seems like something we should handle in mixed domain transformation.

Reply:
If you agree, filing a JIRA, noting it here, and proceeding with this as a temp workaround is fine with me, particularly if this problem is delaying work on https://issues.redhat.com/browse/WFLY-16610.

Reply:
Yes, I also asked on [WFLY-16630](https://issues.redhat.com/browse/WFLY-16630) about it. 

This is not blocking [WFLY-16610](https://issues.redhat.com/browse/WFLY-16610), but I feel more comfortable running these tests without errors since I guess they could affect the test stability. 

If the clustering team agrees to treat it as an infinispan transformer issue, we can remove this workaround together with the infinispan transformer. This workaround removal would become a test case for it. I'm not sure how easy/complex would this transformer be.

Reply:
@bstansberry @yersan I will add an AttributeConverter to handle this.

Reply:
Thanks, @pferraro, I'm going to push this one on top of yours

Reply:
well, there is no need, both can be managed independently